rust/private/rustc.bzl

```diff
@@ -405,7 +405,7 @@ def add_crate_link_flags(args, dep_info):
     )
 
 def _crate_to_link_flag(crate_info):
-    return ["--extern", "{}={}".format(crate_info.name, crate_info.dep.output.path)]
+    return ["--extern", "{}={}".format(crate_info.name.replace("-", "_"), crate_info.dep.output.path)]
```

Comment:
If this replacement always happens, it should be done when the CrateInfo struct is created

Reply:
Maybe this can happen one time, on L105?

Reply:
Good call, updated. That means we no longer need to touch rustc.bzl for this change, only rust.bzl.